samples/matter/common/src/dfu/smp/suit_dfu_over_smp.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "dfu_over_smp.h"
+
+#if !defined(CONFIG_MCUMGR_TRANSPORT_BT) || !defined(CONFIG_SUIT)
+#error "DFUOverSMP requires MCUMGR and SUIT module configs enabled"
+#endif
+
+#include "dfu/ota/ota_util.h"
+
+#include <platform/CHIPDeviceLayer.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+using namespace ::chip;
+using namespace ::chip::DeviceLayer;
+
+namespace Nrf
+{
+
+DFUOverSMP DFUOverSMP::sDFUOverSMP;
+
+void DFUOverSMP::Init()
```

Comment:
I'd consider abstracting the common code to the base class with default implementation of the whole public interface and then only adding different bits in the specific SUIT DFUOverSMP version by overwriting.